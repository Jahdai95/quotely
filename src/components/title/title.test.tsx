import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, test, it, expect } from "vitest";
import Title from "./title";
import { QuotesProvider } from "../../providers/quotes.provider";

describe("Should render Title component", () => {
  const getRender = () => {
    return render(
      <QuotesProvider>
        <Title />
      </QuotesProvider>
    );
  };
  test("Button should click", async () => {
    getRender();

    const button = screen.getByTestId("button-add-quote");
    await userEvent.click(button);
  });

  it("Should macth snapshot", () => {
    const { container } = getRender();
    expect(container).toMatchSnapshot();
  });
});
