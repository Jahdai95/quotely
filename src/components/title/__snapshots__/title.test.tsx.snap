// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render Title component > Should macth snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1kw3y0a"
  >
    <h1
      class="fontSize-lg quicksand-200"
    >
      Citando citas
    </h1>
    <button
      aria-label="Agregar cita"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-14y7ap6-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      data-testid="button-add-quote"
      id=":r7:"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
        />
      </svg>
    </button>
  </div>
</div>
`;
